### PR TITLE
Format embedded GQL in template literal statements

### LINKED ISSUE #16007: Bare embedded GQL string in e.g. JS is not formatted
**Prettier 3.2.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAzCFMC8mA9AFSYDiATgIYAOAFgIoAympxmABgDpSaYAjgFc4VAJ448mYJgCWAE0wBfPlwDcfPmUq1Grdp178hoiVPyzFKtepAAaEBDow56ZKBpUqEAO4AFLwQ0ZBAaABtfGnEQxwAjWjAAazgYAGUaAFs4FjkoOGRsCLQ4eMSU9LoaMDyAc2QYKlFHEsy5BqbSkDgADzoxOWzYCIAVMSgvOTgQpCLwkpa68LgmYQh4QuKugCs0HrSllbWN2a3HEXW4fx86GbC0AFp8uAUXhxBGmjlwuoBhCEymRooQi4XeaEOAEEYI05HFhPB-GJcvlNvMugwYJlwgB1BhyeBoKpgOBpYIEuQANwJ4lCYDQsRAlNEAEkoK9YGkwFQ5C5Iey0jBxMs0QsQHQfCUcXpQhLpmJKQVHHkSlQYNcaLUgaKulUqKrQnEaHE4GDHBK8jAcYoYAxkAAOAAMjiocBEcldGq1wNO6McMGN1oUtuQACZHMISiNjTM5mK4JkTQpXgoWDQoLVhJq4AAxCBUIEwuoghEQEDKZRAA)

```sh
--parser babel
```

**Input:**

```jsx
const foo = /* GraphQL */ `
  query foo { id }
`;

foo(/* GraphQL */ `
  query foo { id }
`);

/* GraphQL */ `
  query foo { id }
`;
```

**Output:**

```jsx
const foo = /* GraphQL */ `
  query foo {
    id
  }
`;

foo(/* GraphQL */ `
  query foo {
    id
  }
`);

/* GraphQL */ `
  query foo { id }
`;
```

**Expected output:**

```jsx
const foo = /* GraphQL */ `
  query foo {
    id
  }
`;

foo(/* GraphQL */ `
  query foo {
    id
  }
`);

/* GraphQL */ `
  query foo {
    id
  }`;
```

**Why?**

We use GQL snippets in TS sometimes purely for codegen - these strings get stripped out during build. We don't need or want to assign to variables, so it would be nice to get these formatted by prettier.

I'd consider this to be a nice-to-have feature. I have no idea if this was a conscious decision, or if it's some kind of design limitation. I'd be willing to implement if this feature would be considered and somebody pointed me in the right direction.

### PR DESCRIPTION
**Pull request by keithlayne**
## Description

Fixes #16007

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by keithlayne** [comment]
My description of this feature is probably bad, feel free to fix :)

**Comment by sosukesuzuki** [comment]
Can you share a real world use case?

**Comment by keithlayne** [comment]
> Can you share a real world use case?

I did in #16007.

**Comment by keithlayne** [comment]
Also maybe worth noting that all my other tooling (codegen, GQL LSP) supports this use case well. It's just a remaining nit that I have to format it by hand right now.

File: src/language-js/embed/utils.js [hunk 1]
```diff
@@ -80,6 +80,9 @@ function hasLanguageComment({ node, parent }, languageName) {
   return (
     hasLeadingBlockCommentWithName(node, languageName) ||
     (isAsConstExpression(parent) &&
+      hasLeadingBlockCommentWithName(parent, languageName)) ||
+    (parent.type === "ExpressionStatement" &&
+      node.type === "TemplateLiteral" &&
       hasLeadingBlockCommentWithName(parent, languageName))
```
Comment (fisker) [inline src/language-js/embed/utils.js:86]:
```suggestion
    (parent.type === "ExpressionStatement" &&
      hasLeadingBlockCommentWithName(parent, languageName))
```

**Review by fisker — APPROVED** [review]

File: src/language-js/embed/utils.js (hunk 1 above)
Comment (keithlayne) [reply src/language-js/embed/utils.js:86]: Thanks - I think I realized that check wasn't strictly necessary, but I just don't know enough about how things work here, so in my mind I was playing it safe somehow. I'll update this today.

Comment (keithlayne) [reply src/language-js/embed/utils.js:86]: Also, would a better description be: `Format embedded GQL in template literal statements`?

**Comment by keithlayne** [comment]
Anything else I need to do here?

**Review by sosukesuzuki — APPROVED** [review]

**Comment by sosukesuzuki** [comment]
Sorry for the delay, will merge when CI goes through.

**Pull request merged**